flocker/provision/_install.py

```diff
@@ -205,11 +206,28 @@ def ensure_minimal_setup(package_manager):
         raise UnsupportedDistribution()
 
 
-def task_cli_pkg_test():
+def cli_pkg_test(package_source=PackageSource()):
     """
-    Check that the CLI is working.
+    Check that the Flocker CLI is working and has the expected version.
+
+    :param PackageSource package_source: The source from which to install the
+        package.
+
+    :return: An ``Effect`` to pass to a ``Dispatcher`` that supports
+        ``Sequence``, ``Run``, ``Sudo``, ``Comment``, and ``Put``.
     """
-    return run_from_args(['flocker-deploy', '--version'])
+    expected = package_source.version
+    if not expected:
+        if package_source.branch:
+            # If branch is set but version isn't, we don't know the
+            # latest version. In this case, just check that the version
+            # can be displayed.
+            return run('flocker-deploy --version')
+        else:
+            # If neither branch nor version is set, the latest
+            # installable release will be installed.
+            expected = get_installable_version(version)
+    return run('test `flocker-deploy --version` = {}'.format(quote(expected)))
```

Comment:
This would quickly get complex. We can get the output from the effects, but need to isolate the line containing the version number. Usually, this would be the last line, but then we need to handle stray warnings from the shell, error messages from the Flocker code, and empty results from the command.